File: services/product-service/app/src/main/java/io/micro_spring_boot/product_service/app/dto/GetProductResponse.java
package io.micro_spring_boot.product_service.app.dto;

import io.micro_spring_boot.product_service.domain.aggregates.product.ProductType;

import java.util.UUID;


public record GetProductResponse(
    UUID id,
    String skuCode,
    String name,
    String description,
    Double price,
    ProductType type
) { }

File: services/product-service/app/src/main/kotlin/com/microspringboot/productservice/app/dtos/GetProductResponse.kt
package com.microspringboot.productservice.app.dtos

import com.microspringboot.productservice.domain.aggregates.products.ProductType
import java.util.*

data class GetProductResponse(
    val id: UUID,
    val skuCode: String,
    val name: String,
    val description: String,
    val price: Double,
    val type: ProductType
)
